src/main/java/mate/academy/spring/controller/MovieSessionController.java

```diff
@@ -0,0 +1,67 @@
+package mate.academy.spring.controller;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.stream.Collectors;
+import mate.academy.spring.dto.MovieSessionRequestDto;
+import mate.academy.spring.dto.MovieSessionResponseDto;
+import mate.academy.spring.model.MovieSession;
+import mate.academy.spring.service.MovieSessionService;
+import mate.academy.spring.service.mapper.MovieSessionMapper;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/movie-sessions")
+public class MovieSessionController {
+    private final MovieSessionService movieSessionService;
+    private final MovieSessionMapper movieSessionMapper;
+
+    public MovieSessionController(MovieSessionService movieSessionService,
+                                  MovieSessionMapper movieSessionMapper) {
+        this.movieSessionService = movieSessionService;
+        this.movieSessionMapper = movieSessionMapper;
+    }
+
+    @PostMapping
+    public MovieSessionResponseDto create(@RequestBody MovieSessionRequestDto
+                                                      movieSessionRequestDto) {
+        return movieSessionMapper.toDto(movieSessionService
+                .add(movieSessionMapper.toModel(movieSessionRequestDto)));
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> getAvailableSessions(@RequestParam Long
+                                                                          movieId,
+                                                              @RequestParam
+                                                              @DateTimeFormat
+                                                                      (pattern = "dd.MM.yyyy")
+                                                                      LocalDate date) {
+        return movieSessionService.findAvailableSessions(movieId, date)
+                .stream()
+                .map(movieSessionMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @DeleteMapping("/{movieSessionId}")
+    public void delete(@PathVariable Long movieSessionId) {
+        movieSessionService.delete(movieSessionId);
+    }
+
+    @PutMapping("/{movieSessionId}")
+    public MovieSessionResponseDto update(@PathVariable Long movieSessionId,
```

Comment:
Follow task requirements 
![image](https://user-images.githubusercontent.com/32339836/235190430-b0ead3d9-8656-4902-8b7a-551676e5939d.png)

```suggestion
    @DeleteMapping("/{id}")
    public void delete(@PathVariable Long id) {
        movieSessionService.delete(movieSessionId);
    }

    @PutMapping("/{id}")
    public MovieSessionResponseDto update(@PathVariable Long id,
```